src/randoop/JunitFileWriter.java

```diff
@@ -105,46 +100,41 @@ private void createOutputDir() {
     return ret;
   }
 
-
-  private File writeSubSuite(List<ExecutableSequence> sequencesForOneFile, String junitTestsClassName) {
+  /*
+   * Writes a JUnit4 test class for a list of sequences
+   */
+  private File writeSubSuite(List<ExecutableSequence> sequences, String testClassName) {
     if (GenInputsAbstract.pretty_print) {
-      SequencePrettyPrinter printer = new SequencePrettyPrinter(sequencesForOneFile, packageName, junitTestsClassName);
+      SequencePrettyPrinter printer = new SequencePrettyPrinter(sequences, packageName, testClassName);
       return printer.createFile(getDir().getAbsolutePath());
     }
 
-    String className = junitTestsClassName;
+    String className = testClassName;
     File file = new File(getDir(), className + ".java");
     PrintStream out = createTextOutputStream(file);
 
     try {
       outputPackageName(out, packageName);
       out.println();
-      out.println("import junit.framework.*;");
+      out.println("import org.junit.Test;");
       out.println();
-      out.println("public class " + className + " extends TestCase {");
+      out.println("public class " + className + " {");
       out.println();
       out.println("  public static boolean debug = false;");
       out.println();
-      int testCounter = 1;
-      for (ExecutableSequence s : sequencesForOneFile) {
+      int testCounter = 0;
+      for (ExecutableSequence s : sequences) {
         if (includeParseableString) {
           out.println("/*");
           out.println(s.sequence.toString());
           out.println("*/");
         }
-        out.println("  public void test" + testCounter + "() throws Throwable {");
-        out.println();
-        // Replaced this printf by the below to avoid a dependence on Java
-        // 5 -- printf was added to the PrintStream class only in J2SE 5.0.
-        // out.println(indent("if (debug) System.out.printf(\"%n" + className + ".test" + testCounter + "\");"));
-        out.println(indent("if (debug) { System.out.println(); System.out.print(\"" + className + ".test" + testCounter + "\"); }"));
-        out.println();
-        out.println(indent(s.toCodeString()));
-        out.println("  }");
-        out.println();
         testCounter++;
```

Comment:
The counter is an artifact of the name generation. I pulled the name generation into an inner class. (Yes, it does the same thing.)